pre_commit_hooks/check_ast.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import absolute_import
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import argparse
+import ast
+import os.path
+import sys
+import traceback
+
+
+def check_ast(argv=None):
+    parser = argparse.ArgumentParser()
+    parser.add_argument('filenames', nargs='*')
+    args = parser.parse_args(argv)
+
+    _, interpreter = os.path.split(sys.executable)
+
+    retval = 0
+    for filename in args.filenames:
+
+        try:
+            ast.parse(open(filename, 'rb').read(), filename=filename)
```

Comment:
not really, I exhaust the entire file immediately (and cpython'll just close the file right there)